iceoryx_hoofs/test/moduletests/test_containers_uninitialized_array.cpp

```diff
@@ -0,0 +1,252 @@
+    auto& buffer = this->buffer;
+    EXPECT_EQ(buffer.begin(), &buffer[0]);
+}
+
+TYPED_TEST(UninitializedArrayTest, ConstBeginReturnsIteratorToBeginningOfUninitializedArray)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "7387b043-db44-47ac-a2da-c40040bb9baa");
+    auto& buffer = this->buffer;
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-const-cast) reuse of buffer to test const method
+    EXPECT_EQ(const_cast<const decltype(buffer)>(buffer).begin(), &buffer[0]);
+}
+
+TYPED_TEST(UninitializedArrayTest, EndReturnsIteratorToEndOfUninitializedArray)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "52447fba-0c7f-40df-8b7f-64d8b3ffcc49");
+    auto& buffer = this->buffer;
+    EXPECT_EQ(buffer.end(), &buffer[buffer.capacity()]);
+}
+
+TYPED_TEST(UninitializedArrayTest, ConstEndReturnsIteratorToEndOfUninitializedArray)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "2946ad83-b782-4c54-966b-c94b482335cc");
+    auto& buffer = this->buffer;
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-const-cast) reuse of buffer to test const method
+    EXPECT_EQ(const_cast<const decltype(buffer)>(buffer).end(), &buffer[buffer.capacity()]);
+}
+
+TEST(UninitializedArrayTest, BeginAndEndIteratorNotEqualInNonEmptyUninitializedArray)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "1a180d15-7e77-4234-ad88-04673cbf9fc9");
+    constexpr uint64_t CAPACITY = 3;
+    using Buffer = UninitializedArray<uint32_t, CAPACITY>;
+    Buffer buffer;
+
+    new (&buffer[0]) Buffer::value_type(1);
+
+    EXPECT_NE(buffer.begin(), buffer.end());
+}
+
+TEST(UninitializedArrayTest, BeginAndEndConstIteratorNotEqualInNonEmptyUninitializedArray)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "33e3d5b4-4762-421a-829f-455fe44b8e3b");
+    constexpr uint64_t CAPACITY = 3;
+    using Buffer = UninitializedArray<int32_t, CAPACITY>;
+    Buffer buffer;
+
+    new (&buffer[0]) Buffer::value_type(2);
+
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-const-cast) reuse of buffer to test const methods
+    EXPECT_NE(const_cast<const Buffer&>(buffer).begin(), const_cast<const Buffer&>(buffer).end());
+}
+
+TYPED_TEST(UninitializedArrayTest, BeginAndEndIteratorNotEqualInFullUninitializedArray)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "2ac41459-1055-47c0-8dc1-ea43c50827bf");
+    auto& buffer = this->buffer;
+    this->fillBuffer(0);
+    EXPECT_NE(buffer.begin(), buffer.end());
+}
+
+TYPED_TEST(UninitializedArrayTest, BeginAndEndConstIteratorNotEqualInFullUninitializedArray)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "01a5d1cd-ba6d-422e-a807-1ffe5787f4af");
+    auto& buffer = this->buffer;
+    this->fillBuffer(2);
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-const-cast) reuse of buffer to test const methods
+    EXPECT_NE(const_cast<const decltype(buffer)>(buffer).begin(), const_cast<const decltype(buffer)>(buffer).end());
+}
+
+TEST(UninitializedArrayTest, IteratorIteratesThroughNonEmptyUninitializedArray)
```

Comment:
I think such an array is never empty (neither is a `std::array`).
```suggestion
TEST(UninitializedArrayTest, IteratorIteratesThroughUninitializedArray)
```